### ISSUE TITLE
Staging a hunk collapses the file

### ISSUE THREAD

**Issue by brunoqc** [magit/magit#2388]
When I stage a chunk from a Go file, the diff view closes. I don't know if it only happens with Go files (it happens if I remove `go-mode`).

I tested with a very small file and I didn't have the problem.

before I press 's'
![image](https://cloud.githubusercontent.com/assets/24027/11095448/ffaa4052-8862-11e5-94d1-e92eb1a4ad1b.png)

after I press 's'
![image](https://cloud.githubusercontent.com/assets/24027/11095472/1c29a1f0-8863-11e5-9ca7-f05654ac25be.png)

test emacs config file
```lisp
(require 'package)
(package-initialize)

(require 'magit)
(require 'go-mode)
```

Magit 20151109.1450, Git 2.6.2.windows.1, Emacs 24.5.1
GNU Emacs 24.5.1 (x86_64-w64-mingw32) of 2015-05-16 on KAEL
Windows 10 64-bit

**Comment by brunoqc** [comment]
I can't test with magit 20151111.418 yet. Maybe I need to wait for git-commit to be updated in melpa

```
package-compute-transaction: Need package `git-commit-20151111', but only 20151028.1203 is available
```

**Comment by halbtuerke** [comment]
I have the exact same problem regardless of the file.

**Comment by tarsius** [comment]
@halbtuerke can you reproduce this with *any* file or any *Go* (the language) [1] file.

[1] Sorry but Go (the game) came earlier, so I call it just Go and Go (the language) is what needs the qualifier.

**Comment by halbtuerke** [comment]
I can reproduce this with any file.

**Comment by npostavs** [comment]
I think this is the effect of `magit-diff-expansion-threshold`. It's not related to the specific file, but rather how many diffs you currently have (so it's more likely to trigger with bigger files).

@brunoqc @halbtuerke can you confirm that doing `(setq magit-diff-expansion-threshold 999.0)` fixes your problem?

**Comment by halbtuerke** [comment]
For every file this behavior occurred, this seems to fix the problem for me. :+1:

**Comment by brunoqc** [comment]
It works for me too.

> It's not related to the specific file, but rather how many diffs you currently have (so it's more likely to trigger with bigger files).

One file I had the problem with only had 2 diff chunks. The file has 1226 lines.

**Comment by tarsius** [comment]
Since the above screenshots shows that the very first file is collapsed and `magit-diff-expansion-threshold` defaults to one second, we know that doing the things that happen before that take more than one second. That's very long, because there's not much going on there. I would also expect that to be somewhat unusual, so I am not increasing the default threshold.

This variable is a hack anyway, and after I have released v2.4 I'll start working on performance. so I am closing this issue.
